### ISSUE TITLE
[Question] What order of quality does DeezLoadr prefer?

### ISSUE THREAD

**Issue by tabepic** [J05HI/DeezLoadr#74]
The function getValidTrackQuality() doesn't seem to have a clear pattern as to what it falls back to if the selected quality isn't available.

**Comment by J05HI** [comment]
FLAC -> 320kbps -> 256kbps -> 128kbps
320kbps -> 256kbps -> 128 -> FLAC
128kbps -> 256kbps -> 320kbps -> FLAC

**Comment by tabepic** [comment]
So... it's not as simple as "use the best quality that's lower than the prefered"?

**Comment by J05HI** [comment]
No it's my definition.
Seems logic for me :)

**Comment by tabepic** [comment]
Okay.
Could there be an option to set the order (when a config file is created) and have "320kbps -> 256kbps -> 128 -> FLAC" as default?

**Comment by J05HI** [comment]
@tabepic 
Here: [https://github.com/J05HI/DeezLoadr/issues/21](https://github.com/J05HI/DeezLoadr/issues/21)

**Comment by tazitus** [comment]
I agree that the getValidTrackQuality() function is kind of ugly and confusing. Something like
```
const order = [musicQualities.FLAC, musicQualities.MP3_320, musicQualities.MP3_256, musicQualities.MP3_128];

function getValidTrackQuality(trackInfos) {
  for(i = order.indexOf(selectedMusicQuality); i < order.length; i++) {
    let filesize = getTrackFilesize(trackInfos, order[i]);
    if !(typeof filesize === 'undefined' || parseInt(filesize) == 0) {
      return quality;
  }
  return false;
}

function getTrackFilesize(trackInfos, quality) {
  switch(quality) {
    case musicQualities.FLAC:
      return trackInfos.FILESIZE_FLAC;
    case musicQualities.MP3_320:
      return trackInfos.FILESIZE_MP3_320;
    case musicQualities.MP3_256:
      return trackInfos.FILESIZE_MP3_256;
    case musicQualities.MP3_128:
      return trackInfos.FILESIZE_MP3_128;
  }
}
```
would get the job done while being easier to understand, debug, etc. and providing a variable order of quality preferences.

**Comment by tabepic** [comment]
@tazitus That would definitely help

What I'm wondering is how we could present the re-orderable list to the user; AFAIK inquirer doesn't really support this (at least not without recreating a menu everytime a change is made).

**Comment by J05HI** [comment]
Wait for the GUI :)

**Comment by J05HI** [comment]
@tazitus 
Send a pull request :)

**Comment by J05HI** [comment]
@tazitus 
Could you please send a pull request :)

**Comment by tabepic** [comment]
@J05HI I'd assume it's easy to put in a GUI, I was thinking that this would be implemented in the CLI though.

**Comment by J05HI** [comment]
@tabepic
Not reordering

**Comment by J05HI** [comment]
# Project moved!
## New project: [https://github.com/J05HI/SMLoadr](https://github.com/J05HI/SMLoadr)
## Please delete all of your forks!!!
## I'll delete this repo soon!
